src/lib/mathlib/math/filter/LowPassFilter2pVector3f.hpp

```diff
@@ -0,0 +1,89 @@
+/****************************************************************************
+ *
+ *   Copyright (C) 2018 PX4 Development Team. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/// @file	LowPassFilter2pVector3f.hpp
+/// @brief	A class to implement a second order low pass filter on a Vector3f
+/// Author: Leonard Hall <LeonardTHall@gmail.com>
+/// Based on LowPassFilter2p.hpp
+
+#pragma once
+
+#include <matrix/math.hpp>
+
+namespace math
+{
+class LowPassFilter2pVector3f
+{
+public:
+
+	LowPassFilter2pVector3f(float sample_freq, float cutoff_freq)
+	{
+		// set initial parameters
+		set_cutoff_frequency(sample_freq, cutoff_freq);
+	}
+
+	// Change filter parameters
+	void set_cutoff_frequency(float sample_freq, float cutoff_freq);
+
+	// Add a new raw value to the filter
+	//
+	// @return retrieve the filtered result
```

Comment:
- can you change to the typical comments we use `/**`...?
- can you add an explicit `inline` (I know that it's already implicitly requested to be inlined)?

Reply:
This is the kind of thing where we need a simple style guide or even tool to enforce. I've generally been moving towards c++ style comments, but don't feel strongly about it. I think even doxygen could be used to raise an error.